### PR TITLE
[plug-in][debug]Make it possible to connect to remote debug server

### PR DESCRIPTION
**Pull request by svor**
Signed-off-by: Valeriy Svydenko <vsvydenk@redhat.com>

This PR provides an ability to connect to remote debug server if it was provided by debug configuration.

### CODE REVIEW

File: packages/plugin-ext/src/plugin/node/debug/debug.ts [hunk 1]
```diff
@@ -182,8 +182,13 @@ export class DebugExtImpl implements DebugExt {
     }
 
     async $createDebugSession(debugConfiguration: theia.DebugConfiguration): Promise<string> {
-        const executable = await this.getExecutable(debugConfiguration);
-        const communicationProvider = startDebugAdapter(executable);
+        let communicationProvider: CommunicationProvider;
+        if ('debugServer' in debugConfiguration) {
+            communicationProvider = connectDebugAdapter(debugConfiguration.debugServer);
```
Comment (benoitf) [inline packages/plugin-ext/src/plugin/node/debug/debug.ts]: debugServer is just a port ? not a hostname and a port ?

Comment (svor) [reply packages/plugin-ext/src/plugin/node/debug/debug.ts]: just a port

Comment (svor) [reply packages/plugin-ext/src/plugin/node/debug/debug.ts]: At least I don't know cases where it should be `hostname:port`

Comment (tolusha) [reply packages/plugin-ext/src/plugin/node/debug/debug.ts]: yes, in `vscode` it is just a port
https://github.com/Microsoft/vscode/blob/master/src/vs/workbench/parts/debug/node/debugger.ts#L272

Comment (benoitf) [reply packages/plugin-ext/src/plugin/node/debug/debug.ts]: @tolusha yes thx, saw it there as well: https://code.visualstudio.com/docs/editor/debugging

Comment (svor) [reply packages/plugin-ext/src/plugin/node/debug/debug.ts]: @tolusha thanks

**Pull request merged**